### PR TITLE
Collect targets in a deterministic order

### PR DESCRIPTION
**Pull request by dtolnay**
Fixes the nondeterminism pointed out in https://github.com/bazelbuild/rules_rust/pull/1735#pullrequestreview-1233384845.

### CODE REVIEW

File: crate_universe/src/context/crate_context.rs [hunk 1]
```diff
@@ -28,6 +28,12 @@ pub struct CrateDependency {
 #[serde(default)]
 pub struct TargetAttributes {
     /// The module name of the crate (notably, not the package name).
+    ///
+    /// This must be the first field of `TargetAttributes` to make it the
```
Comment (UebelAndre) [inline crate_universe/src/context/crate_context.rs:32]: Small nit but I don't think this information is ideal for docs. Feels like just a code comment. Do you know of a good way to not have this show up in rustdoc renderings?

**Review by UebelAndre — APPROVED** [review]
Thank you! 

I had one nit I was more curious if you'd come across a pattern for.

File: crate_universe/src/context/crate_context.rs (hunk 1 above)
Comment (dtolnay) [reply crate_universe/src/context/crate_context.rs]: We can use `//` comments.

**Comment by UebelAndre** [comment]
Good to merge after a rebase 😄

**Comment by dtolnay** [comment]
Rebased.

**Comment by dtolnay** [comment]
Failure in "Crate Universe Unnamed Examples on Windows (OpenJDK 11, VS2019)" appears unrelated.

```console
(18:42:34) ERROR: C:/b/rfwvwffm/external/rust_windows_x86_64__x86_64-pc-windows-msvc__stable_tools/BUILD.bazel:89:15: Creating symlink bazel-out/x64_windows-fastbuild/bin/external/rust_windows_x86_64__x86_64-pc-windows-msvc__stable_tools/rust_toolchain/lib/rustlib/x86_64-pc-windows-msvc/lib/std-e33da769ef0fc5d8.dll failed: failed to create symbolic link 'bazel-out/x64_windows-fastbuild/bin/external/rust_windows_x86_64__x86_64-pc-windows-msvc__stable_tools/rust_toolchain/lib/rustlib/x86_64-pc-windows-msvc/lib/std-e33da769ef0fc5d8.dll' to 'external/rust_windows_x86_64__x86_64-pc-windows-msvc__stable_tools/lib/rustlib/x86_64-pc-windows-msvc/lib/std-e33da769ef0fc5d8.dll' due to I/O error: C:/b/rfwvwffm/execroot/__main__/bazel-out/x64_windows-fastbuild/bin/external/rust_windows_x86_64__x86_64-pc-windows-msvc__stable_tools/rust_toolchain/lib/rustlib/x86_64-pc-windows-msvc/lib/std-e33da769ef0fc5d8.dll (Permission denied)
```

**Comment by dtolnay** [comment]
It passed after a retry.

**Pull request merged**